docs/resources/devurl.md

```diff
@@ -0,0 +1,63 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "coder_devurl Resource - terraform-provider-coder"
+subcategory: ""
+description: |-
+  Use this resource to define shortcuts to access applications in a workspace.
+---
+
+# coder_devurl (Resource)
+
+Use this resource to define shortcuts to access applications in a workspace.
+
+## Example Usage
+
+```terraform
+resource "coder_agent" "dev" {
+  os             = "linux"
+  arch           = "amd64"
+  dir            = "/workspace"
+  startup_script = <<EOF
+curl -fsSL https://code-server.dev/install.sh | sh
+code-server --auth none --port 13337
+EOF
+}
+
+resource "coder_devurl" "code-server" {
+  agent_id = coder_agent.dev.id
+  name     = "VS Code"
+  icon     = "builtin:vscode"
+  target   = "http://localhost:13337"
+}
+
+resource "coder_devurl" "vim" {
+  agent_id = coder_agent.dev.id
+  name     = "Vim"
+  icon     = "builtin:vim"
+  command  = "vim"
+}
+
+resource "coder_devurl" "intellij" {
+  agent_id = coder_agent.dev.id
+  icon     = "builtin:intellij"
+  name     = "JetBrains IntelliJ"
+  command  = "projector run"
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `agent_id` (String) The "id" property of a "coder_agent" resource to associate with.
+
+### Optional
+
+- `command` (String) A command to run in a terminal opening this DevURL. In the web, this will open in a new tab. In the CLI, this will SSH and execute the command.
+- `icon` (String) An icon to display in the dashboard.
```

Comment:
This doesn't really explain how to use the `icon`. I see the examples all use `builtin:x` strings, is there a way to see all builtin icons? Is it possible to use a non-builtin icon?

Reply:
```
provider.coder.url + "/icons/goland.png"
```

Reply:
```
data.coder_workspace.me.external_url
```